app/controllers/api/base_controller.rb

```diff
@@ -13,5 +13,16 @@ class BaseController < Spree::Api::BaseController
     def respond_with_conflict(json_hash)
       render json: json_hash, status: :conflict
     end
+
+    private
+
+    # Use logged in user (spree_current_user) for API authentication (current_api_user)
+    def authenticate_user_with_spree_current_user
+      @current_api_user = try_spree_current_user
+      
+      # Authenticate API user
+      authenticate_user_without_spree_current_user
+    end
+    alias_method_chain :authenticate_user, :spree_current_user
```

Comment:
Although this solves the problem, IMO there's no benefit on using `alias_method_chain` here. There are two things that lead me to think we should implement this differently:

* We've been actively removing uses of `alias_method_chain` in the upgrade and this goes in the opposite direction.
* It's been removed from Rails codebase itself and it's somewhat discouraged in the Rails community, for the reasons explained in https://yehudakatz.com/2009/03/06/alias_method_chain-in-models/ and https://www.justinweiss.com/articles/rails-5-module-number-prepend-and-the-end-of-alias-method-chain/.

If I'm not mistaken we can leverage the inheritance we're already using here. What about?

```suggestion
    def authenticate_user
      @current_api_user = try_spree_current_user
      super
    end
```

Reply:
ah, yes, we are extending not class_evaling.... this was a SUPER suggestion @sauloperez :-)

Reply:
lol @luisramos0